roles/common/tasks/hostname.yml

```diff
@@ -0,0 +1,59 @@
+---
+# roles/hostname/tasks/main.yml
+# The _domain variable functions as a FQDN to handle deployments using a subdomain, top level domain and www subdomain, or default rDNS. 
+# When domain is defined
+- name: set dns fact for subdomain
+  set_fact:
+    _domain: '{{ subdomain }}.{{ domain }}'
+    cacheable: yes
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+  when:
+    - domain is defined
+    - subdomain != "www"
+    
+- name: set dns fact for domain
+  set_fact:
+    _domain: '{{ domain }}'
+    cacheable: yes
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+  when: 
+    - domain is defined
+    - subdomain == "www"
+  
+- name: setting system hostname on master
+  hostname:
+    name: "{{ _domain }}" 
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+  when: 
+    - domain is defined
+
+# Since worker UIs are viewed through reverse proxy, hostname will always be default dns on workers
+- name: setting system hostname on workers
+  hostname:
+    name: "{{ ansible_default_ipv4.address | replace('.', '-') }}.ip.linodeusercontent.com"
+  when: 
+    - domain is defined
+    - "'spark_workers' in group_names"
+
+# When domain is not defined
+- name: set dns fact for default rdns
+  set_fact:
+    _domain: "{{ ansible_default_ipv4.address | replace('.', '-') }}.ip.linodeusercontent.com"
+    cacheable: yes
+  when: domain is not defined
+
+- name: setting system hostname
+  hostname:
+    name: "{{ _domain }}" 
+  when: domain is not defined
+
+- name: add hostname to /etc/hosts
+  lineinfile:
+    path: /etc/hosts
+    line: "{{ ansible_default_ipv4.address }} {{ _domain }}"
+    insertafter: EOF
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
```

Comment:
We should be updating the hosts file on all nodes. The hosts file should have every node in the cluster

Reply:
Done.